### PR TITLE
[HEI-6] [FEATURE] Add new dataset.py and related functions

### PR DESCRIPTION
**Pull request by tikhomirovd**
## 🚀 Feature Proposal

Introduce a new module, `dataset.py`, containing several synthetic data generators to facilitate testing and experimentation within the HypEx library.

### Motivation

To enhance the capabilities of HypEx in simulating various data-driven scenarios, enabling developers and researchers to better understand and demonstrate the functionalities of HypEx.

### Feature Description

The `dataset.py` module will include functions such as `gen_special_medicine_df`, `gen_oracle_df`, and `gen_control_variates_df`. These functions will generate synthetic datasets based on specified parameters, supporting the development and testing of data analysis methods in HypEx.

### Potential Impacts

The introduction of `dataset.py` will primarily impact the testing and demonstration aspects of HypEx, with minimal performance implications on the core functionalities.

### Alternatives

While external data generation tools could be used, having built-in functions ensures seamless integration and consistency in data handling.

### Additional Context

These functions will utilize numpy and pandas libraries to manipulate and generate data, ensuring compatibility with the existing HypEx data structures.

### Checklist

- [x] I have clearly described the feature.
- [x] I have outlined the motivation for the proposal.
- [x] I have provided a detailed description of the feature.
- [x] I have discussed potential impacts and alternatives.
- [x] I have added any additional context or screenshots.

### CODE REVIEW

File: hypex/utils/datasets.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        disease_degree=disease_degree,
+        experimental_treatment=experimental_treatment,
+        residual_lifetime=residual_lifetime,
+    ))
+    return df
+
+
+def gen_oracle_df(
+        data_size=8,
+        *,
```
Comment (VasilievaKA) [inline hypex/utils/datasets.py:93]: ?

File: hypex/utils/datasets.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        Treatment=treatment,
+        Target=y_factual,
+        TE=treatment_effect,
+    ))
+    return df
+
+
+def gen_control_variates_df(
+        data_size=1000,
+        *,
```
Comment (VasilievaKA) [inline hypex/utils/datasets.py:161]: ?

**Review by VasilievaKA — COMMENTED** [review]
Good)

**Review by Dmatryus — APPROVED** [review]
🚀

**Pull request merged**